services/provider/server/cephblockpool.go

```diff
@@ -0,0 +1,112 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"slices"
+
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type cephBlockPoolManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newCephBlockPoolManager(cl client.Client, namespace string) (*cephBlockPoolManager, error) {
+	return &cephBlockPoolManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (c *cephBlockPoolManager) EnableBlockPoolMirroring(ctx context.Context, blockPoolName string) error {
+
+	cephBlockPool := &rookCephv1.CephBlockPool{}
+	cephBlockPool.Name = blockPoolName
+	cephBlockPool.Namespace = c.namespace
+
+	_, err := ctrl.CreateOrUpdate(ctx, c.client, cephBlockPool, func() error {
+		cephBlockPool.Spec.Mirroring.Enabled = true
+		cephBlockPool.Spec.Mirroring.Mode = "image"
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed to enable mirroring on CephBlockPool resource with name %q. %v", blockPoolName, err)
+	}
+
+	return nil
+
+}
+
+func (c *cephBlockPoolManager) SetBootstrapSecretRef(ctx context.Context, blockPoolName string, secretData map[string][]byte) error {
+
+	cephBlockPool, err := c.GetBlockPoolByName(ctx, blockPoolName)
+	if err != nil {
+		return err
+	}
+
+	// create the secret
+	secretJSON, err := json.Marshal(secretData)
+	if err != nil {
+		return err
+	}
+	secretNameMd5Sum := md5.Sum(secretJSON)
+	secretName := hex.EncodeToString(secretNameMd5Sum[:16])
```

Comment:
Instead of deriving a name from the secretInforamion, we could use md5Sum of `blockPoolName-<ClusterID>` which will be passed as a parameter to the grpc req

Reply:
this piece of code is removed, isn't it?

Reply:
Yes it is removed